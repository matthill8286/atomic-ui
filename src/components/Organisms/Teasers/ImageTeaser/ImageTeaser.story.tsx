import { number, radios } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { BadgeActionType, BadgeType } from '@/components/Atoms/Badge/Badge.interface'
import { StorybookRouterWrapper } from '@/utils/StorybookWrapper'
import { ImageTeaser } from './ImageTeaser'
import { ImageTeaserFCProps } from './ImageTeaser.interface'
import { getTransformedImageVersion } from '@/utils/useGraphCmsImages'

const actionTypes: { [key: string]: BadgeActionType } = {
  prominent: 'prominent',
  secondary: 'secondary',
}

const badges: BadgeType[] = [
  { id: 0, name: 'Featured' },
  { id: 1, name: 'Sponsored', actionType: 'prominent' },
]

storiesOf('Design System/Organisms/Teasers/ImageTeaser', module).add('Default', () => {
  const knobs = (): ImageTeaserFCProps => {
    return {
      badgeActionType: radios('Type', actionTypes, actionTypes['prominent']),
      finalImages: [
        getTransformedImageVersion('c8GLaIBVRa6EelN5baVr'),
        getTransformedImageVersion('OWr5QtrXQ1eHiwMdrDM7'),
        getTransformedImageVersion('fj7dbi1rSa2cGlaQqwDG'),
        getTransformedImageVersion('NPgqQFqR0uJ1XhB00sGg'),
      ],
    }
  }

  const height = number('height', 300, {
    range: true,
    min: 100,
    max: 600,
    step: 1,
  })

  return (
    <StorybookRouterWrapper>
      <div
        style={{
          height,
          width: '100%',
          display: 'flex',
        }}>
        <ImageTeaser key={'ImageTeaser_' + height} {...knobs()} badges={badges} />
      </div>
    </StorybookRouterWrapper>
  )
})
