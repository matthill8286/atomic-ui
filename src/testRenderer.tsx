import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
