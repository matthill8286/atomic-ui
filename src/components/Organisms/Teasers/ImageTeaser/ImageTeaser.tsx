import React from 'react'
import { Card } from '@/components/Atoms/Card'
import { Picture } from '@/components/Atoms/Picture'
import { ImageTeaserFCProps } from './ImageTeaser.interface'

export const ImageTeaser: React.FC<ImageTeaserFCProps> = ({
  badges,
  badgeActionType,
  elevation = 0,
  elevationHover = 4,
  finalImages,
}) => {
  return (
    <Card
      elevation={elevation}
      elevationHover={elevationHover}
      shape="rounded-small"
      display="flex"
      fullHeight
      badges={badges}
      badgeActionType={badgeActionType}>
      <Picture
        src={finalImages[0]}
        srcLg={finalImages[1]}
        srcMd={finalImages[2]}
        srcSm={finalImages[3]}
        objectFit="cover"
        height="100%"
        width="100%"
      />
    </Card>
  )
}
