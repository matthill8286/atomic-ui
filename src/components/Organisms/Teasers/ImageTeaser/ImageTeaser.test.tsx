import React from 'react'
import { renderWithThemeAndRouter } from '@/testRenderer'
import { ImageTeaser } from './ImageTeaser'

const images = [
  'public/images/teasers/500x560.jpg',
  'public/images/teasers/700x560.jpg',
  'public/images/teasers/1000x360.jpg',
  'public/images/teasers/1200x360.jpg',
]

describe('ImageTeaser', () => {
  it('renders correctly: default teaser with badges', () => {
    const tree = renderWithThemeAndRouter(
      <ImageTeaser
        badges={[
          { id: 0, name: 'Featured' },
          { id: 1, name: 'Live@@@' },
        ]}
        badgeActionType="prominent"
        finalImages={images}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
