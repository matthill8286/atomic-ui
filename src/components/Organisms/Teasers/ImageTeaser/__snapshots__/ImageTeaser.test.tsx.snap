// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageTeaser renders correctly: default teaser with badges 1`] = `
.c5 {
  margin: 0;
  font-size: 12px;
  line-height: 16px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 {
  background-color: #00C8DC;
  color: #FFFFFF;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 24px;
  padding-right: 24px;
  margin-left: 0;
  margin-right: 9px;
  box-sizing: content-box;
  height: 28px;
  overflow: auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  height: 30px;
}

.c4 {
  display: inline-block;
  position: relative;
  margin-right: 8px;
  max-width: 100%;
}

.c7 {
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c9 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  max-width: 100%;
}

.c9 img {
  height: 100%;
  object-fit: cover;
}

.c2 {
  position: absolute;
  left: 0;
  top: -14px;
  max-width: 100%;
  z-index: 1;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  line-height: 1.43;
  overflow: hidden;
  color: #000000;
  border: 1px solid #E5E5E5;
  background-color: #FFFFFF;
  cursor: pointer;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-property: border-color,box-shadow,-webkit-transform;
  -webkit-transition-property: border-color,box-shadow,transform;
  transition-property: border-color,box-shadow,transform;
  border-radius: 4px;
  height: 100%;
  width: 100%;
}

.c0 {
  position: relative;
  outline: none;
  width: 100%;
  height: 100%;
}

.c0 .c1 {
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c9 {
    max-width: 100%;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c9 img {
    max-width: 100%;
    width: auto;
    height: auto;
  }
}

@media (max-width:752px) {
  .c2 > div > div:nth-child(n + 3) {
    display: none;
  }
}

@media (min-width:512px) {
  .c8:hover {
    -webkit-transform: translateY(-1px);
    -ms-transform: translateY(-1px);
    transform: translateY(-1px);
    box-shadow: 0 24px 38px 3px rgba(0,0,0,0.07),0 9px 46px 8px rgba(0,0,0,0.06),0 11px 15px -7px rgba(0,0,0,0.10);
    border-color: rgba(0,0,0,0);
  }
}

@media (min-width:512px) {
  .c0:hover .c1,
  .c0:hover .Cardstyled__StyledFeaturedAssetsContainer-sc-1gx6c15-3 {
    -webkit-transform: translateY(-1px);
    -ms-transform: translateY(-1px);
    transform: translateY(-1px);
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <span
          className="c5 c6"
          fontSize="xxs"
        >
          <div
            className="c7"
          >
            Featured
          </div>
        </span>
      </div>
      <div
        className="c4"
      >
        <span
          className="c5 c6"
          fontSize="xxs"
        >
          <div
            className="c7"
          >
            Live@@@
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    className="c8"
    display="flex"
    elevation={0}
    shape="rounded-small"
  >
    <div
      className="c9"
      height="100%"
      width="100%"
    >
      <picture
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <source
          media="(max-width: 511px)"
          srcSet="public/images/teasers/1200x360.jpg"
        />
        <source
          media="(max-width: 751px)"
          srcSet="public/images/teasers/1000x360.jpg"
        />
        <source
          media="(max-width: 1007px)"
          srcSet="public/images/teasers/700x560.jpg"
        />
        <img
          alt="image"
          height="100%"
          loading="eager"
          src="public/images/teasers/500x560.jpg"
          width="100%"
        />
      </picture>
    </div>
  </div>
</div>
`;
